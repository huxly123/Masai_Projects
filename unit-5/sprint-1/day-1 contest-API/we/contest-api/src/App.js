import { useContext, useState } from 'react';
import './App.css';
import First from './components/First';
 import {Authcontext} from "./Context/Authcontext"

function App() {

  const [age]=useState(10)
  
  const { token, handletoChange } = useContext(Authcontext);
  
  if (!token) {
    return (
      <>
        <h1>provide token</h1>
        <input type="text"></input>
        <input type="password"></input>
        <button onClick={() => {
          handletoChange("QpwL5tke4Pnpja7X4");
        }}>Submit</button>
      </>
    );
  }
  return (
    <div className="App">
      hello ,your age is:{age}
   <First/>
    </div>
  );
  
}

export default App;
