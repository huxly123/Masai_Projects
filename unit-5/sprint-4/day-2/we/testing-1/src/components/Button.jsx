import React from 'react'

function Button({label,onclick}) {
    return (
        <div>
            <button onclick={onclick}>{label}</button>
        </div>
    )
}

export default Button
