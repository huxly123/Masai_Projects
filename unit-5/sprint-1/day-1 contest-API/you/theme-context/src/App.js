import { useContext } from 'react';
import './App.css';
import { Themecontext } from './Context/ThemeContext';
import {Div} from "./Custom/Div"


function App() {

const {theme,handleTheme}=useContext(Themecontext)

  return (
    <div className="App">
      <Div theme={theme}>
        <button onClick={() => {
          handleTheme("dark")
        }}>Dark</button>
        <button onClick={() => {
          handleTheme("light")
        }}>Light</button>
     </Div>
      

    </div>
  );
}

export default App;
