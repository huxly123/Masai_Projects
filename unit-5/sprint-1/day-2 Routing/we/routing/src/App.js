import "./App.css";
import { Switch, Route, Link } from "react-router-dom";
import UsersList from "./components/UsersList";
import UserDetail from "./components/UserDetail";

function App() {
  return (
    <div className="App">
      <Link to="/">home</Link>
      <Link to="/about">about</Link>
      <Link to="/contact">Contact</Link>
      <Link to="/users">Users</Link>
      <Switch>
        <Route path="/" exact>
          <h1>home</h1>
        </Route>

        <Route path="/about" exact>
          <h1>About</h1>
        </Route>

        <Route path="/contact">
          <h1>contact us:</h1>
          <p>chennai</p>
        </Route>

        <Route path="/users" exact>
          <UsersList />
        </Route>

        <Route path="/user/:id" exact>
        <UserDetail/>
        </Route>

        <Route>404not found</Route>
      </Switch>
    </div>
  );
}

export default App;
