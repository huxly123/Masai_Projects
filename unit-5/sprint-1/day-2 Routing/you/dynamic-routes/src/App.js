
import './App.css';
import {Route,Switch} from "react-router-dom"
import Navbar from './components/Navbar/Navbar';
import Allproducts from './components/All_products/All_products';
import Product from './components/Product';

function App() {
  return (
    <div className="App">
      <Navbar/>
      <Switch>
        <Route path="/" exact>Home</Route>

        <Route path="/products" exact><Allproducts/></Route>

        <Route path="/products/:id" exact><Product/></Route>

        <Route>Home</Route>
      </Switch>
    </div>
  );
}

export default App;
