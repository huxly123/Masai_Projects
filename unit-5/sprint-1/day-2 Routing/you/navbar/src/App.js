import logo from './logo.svg';
import './App.css';
import {
  Route,
  Switch
} from "react-router-dom"
import Navbar from './components/Navbar';

function App() {
  return (
    <div className="App">
      <Navbar />
      <Switch>
        <Route path="/" exact>
          Home
        </Route>

        <Route path="/contact" exact>
          Contact
        </Route>

        <Route path="/about-us" exact>
          About
        </Route>

        <Route path="/services" exact>
          Services
        </Route>

        <Route path="/login" exact>
          Login
        </Route>

        <Route>404: NOT FOUND</Route>
      </Switch>
    </div>
  );
}

export default App;
