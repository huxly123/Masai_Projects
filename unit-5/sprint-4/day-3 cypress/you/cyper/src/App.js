import logo from './logo.svg';
import './App.css';
import Input from './components/Input';
import Toggle from './components/Toggle';

function App() {
  return (
    <div className="App">
      {/* <Input/> */}
      <Toggle/>
    </div>
  );
}

export default App;
