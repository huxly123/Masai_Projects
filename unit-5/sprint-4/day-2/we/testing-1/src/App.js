import './App.css';
import Button from './components/Button';
import Tod from './components/Tod';

function App() {
  return (
    <div className="App">
      {/* <Button/> */}
      <Tod/>
    </div>
  );
}

export default App;
