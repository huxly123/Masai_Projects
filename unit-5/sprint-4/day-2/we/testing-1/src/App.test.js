import { fireEvent, render, screen } from '@testing-library/react';
import App from './App';
import Button from "./components/Button"

xtest('renders learn react link', () => {
  render(<App />);
 
  const linkElement = screen.getByText(/learn react/i);

  expect(linkElement).toBeInTheDocument();

});

 xtest("button to be in document", function () {
  const { getByText } = render(<Button label="click me" />)
  const button = getByText("click me")
  expect(button).toBeInTheDocument();
 

})

xtest("button snap", function () {
  const { container } = render(<Button label="click me" />)
  expect(container).toMatchSnapshot()
})

xtest("button clicks", function () {

const mockFunction=jest.fn()

const {container}=  render(<Button label="click" onclick={() => {
    const myElem = document.createElement("div");
    myElem.textContent = "dummy";
    document.body.appendChild(myElem)
  }} />)

  fireEvent.click(screen.getByText("click"));
  
  //test

  const melement = document.getElementById("dummy");
  expect(melement).toHaveTextContent("dummy");
  expect(melement).toBeInTheDocument()



})





xtest("button clicks function", function () {
  const mockFunction = jest.fn();

  const { container } = render(
    <Button
      label="click"
      onclick={() => 
        mockFunction()
      }
    />
  );

  fireEvent.click(screen.getByText("click"));

  //test

  expect(mockFunction).toBeCalled();
});