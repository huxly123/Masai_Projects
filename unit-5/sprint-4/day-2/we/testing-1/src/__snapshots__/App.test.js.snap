// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button snap 1`] = `
<div>
  <div>
    <button>
      click me
    </button>
  </div>
</div>
`;
